### ISSUE TITLE
Annotator: A Generic Active Learning Baseline for LiDAR Semantic Segmentation

### ISSUE THREAD

**Issue by zhuoxiao-chen** [zhuoxiao-chen/please_read_paper_every_day#48]
## Observation and Motivation

<img width="785" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/2fae83cd-7c63-444f-95e6-1d6f3dcd41c7">

### Cold Start Problem and ASFDA
- The lack of prior information to guide the initial selection of annotated data.
- Access an auxiliary model via pre-training on the open-access auxiliary (source) dataset.
- This auxiliary model serves as a warm-up stage, allowing for smart target data selection for initial annotation.
- ASFDA: active source-free domain adaptation.

### ADA
- Active Domain Adaptation: A labeled auxiliary dataset is available, and the objective is to select target instances for annotating and learn a model with higher segmentation performance on the target test set.

### A voxel-centric selection strategy
- An input LiDAR scene is ﬁrst voxelized into **voxel grids**, with **a large voxel size** to expand the local areas during the selection process.
- After obtaining ﬁnal network predictions, **importance estimation is carried out for each voxel grid** using several common strategies: random, entropy, margin ...
- Propose voxel confusion degree (VCD), which takes into account **nearby predictions**, capturing diversity and redundancy within a voxel grid.
- VCD considers local topology relations, both uncertainty and diversity of a voxel grid. 
- In each active round, we query the top one voxel grid within each scan for annotation until the budget is exhausted.
- It achieves performance on par with the fully-supervised counterpart requiring 1000x fewer annotations

## Methodology

<img width="1111" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/fe437d3f-cdbf-40fb-bfe9-0b322b133cae">

### A generalist Annotator
- Voxelization to get voxel grids
- Active function with online selection for picking **the most valuable voxel grid of each input scan in each active round** 

**ASFDA**, we begin by pre-training a network on the source domain through standard supervised learning. This warm-up network then serves as a strong initialization to aid the initial selection.

**ADA**, except for the pre-training stage, we also **make use of annotated source domain** to promote the selection in each round and facilitate domain alignment.

### The VCD Strategy

<img width="400" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/bac9945d-8206-4db7-9a23-269c698e7d78">

**Equation in my paper**:
<img width="400" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/d7b3a6e0-3544-4a9e-9236-df981d870445">

- Voxel grid with the highest VCD score is selected in each scan.
- A higher score indicates a greater category diversity within a voxel.

### Optimization 
<img width="1268" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/2d0c0f05-5d8b-4ecc-a7fe-27d7c08bbd12">

## Experiments 

<img width="734" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/d5ad87d2-34ce-41f9-bb0e-2869fc09cb11">

<img width="1417" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/374ef923-c36d-4b5b-b552-90f4206d1185">

<img width="938" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/70249f63-cec9-4f46-8411-210d6a52eff7">

<img width="926" alt="image" src="https://github.com/zhuoxiao-chen/please_read_paper_every_day/assets/50613939/dacc10fd-c939-458d-8966-5a222687d78b">

## Notes

1. This is a baseline paper, no complicated and novel method is proposed, the VCD is simple but effective (exactly same as stage 1 of CRB). The goal of this paper is to propose 3 settings: AL, ADA, ASFDA for 3D semantic segmentation, and become a baseline for future work.
2. Although the VCD is too simple, its effectiveness achieved very good performance when using only 0.1% annotation. This is attracting for paper writing. 
3. Although the experiments are extensive, as a baseline paper, it should compare with more AL baselines, not just 3 very basic ones. 
4. The presentation and writing are pretty good. 
5. It is much better than cvpr24 papers I reviewed.
